### DISCUSSION TITLE
How to find a page id? How to list all pages? What are blocks?

### DISCUSSION THREAD

**Discussion by ChrisDelClea** [ramnes/notion-sdk-py#47] [Q&A] [answered]
Hi Guys,

this is more of a question i have. I got started and tried the one example to display the users.
That worked fine. However, after that i tried to find a notion page by it's id. 
So my first question, where to find that `page_id` ? I thought about it's the hash value on the link
of the notion page (however, in the example it includes `-` and in the hash not. 

After that, i just wanted to list all pages that are available, how to do that? 
I could not find anything like `notion.pages.list()` . 

Last but not least, and more of a general understanding question what are blocks? 
How to see what blocks are saved? 

Sorry if this questions are to simple for you, peeps. 

Cheers
Chris

**Comment by aahnik** [comment, marked as answer]
> So my first question, where to find that page_id?

Read this https://github.com/ramnes/notion-sdk-py/discussions/31 and also [this](https://ramnes.github.io/notion-sdk-py/reference/helpers/#notion_client.helpers.get_id).

> I just wanted to list all pages that are available, how to do that?

As far as I know, the official API does not allow that. Listing is available for users and databases. The official pages endpoint offers 3 features - retrieve, create, update. (at the time when this answer is being written)

> what are blocks?

From Notion HQ: “A block is any single piece of content you add to your page, like a to-do item, an image, a code block, an embedded file, etc. Think of your page as being made up of these building blocks.”

**Reply by ChrisDelClea** [reply]
Thank you, @aahnik.

**Reply by aahnik** [reply]
> Thank you, @aahnik.

Hi @ChrisChross, if my answer helped, please mark it as an answer 😊

**Reply by wangdong2023** [reply]
Hi, @aahnik what's the closest way to list pages then? Via search API?

**Comment by Zezombye** [comment]
A page is also a block, so if you have a parent page and you want the child pages, you can use the `https://api.notion.com/v1/blocks/{block_id}/children` endpoint, using the page id as the block id.
